core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -178,10 +192,22 @@ private[chisel3] trait HasId extends InstanceId {
   }
 
   /** This resolves the precedence of [[autoSeed]] and [[suggestName]]
+    *
+    * @note It will become an error in the future to suggestName the same thing that autoSeed would have assigned
     *
     * @return the current calculation of a name, if it exists
     */
-  private[chisel3] def seedOpt: Option[String] = suggested_seed.orElse(auto_seed)
+  private[chisel3] def seedOpt: Option[String] = {
+    suggested_seed.zip(auto_seed).foreach {
+      case (suggested, auto) =>
+        if (suggested == auto) {
+          Builder.deprecated(
+            "calling suggestName(\"" + suggested + "\") had no effect as it is the same as the automatically given name, this will become an error in 3.6"
```

Comment:
Sorry @jackkoenig your long comment here got rebased away (about source locators).

Suggestions on either of these points?:

A.  I can't put `$this` into the debug string because it then becomes recursive call since this is part of toString.
B. I'm not sure how to do what you are saying about make sure the source locator does not show up... how can I suppress a source locator?

Reply:
You could do one of the following to get rid of the source locator: (a) Use `Builder.warningNoLoc`, (b) Use `Builder.deprecated(msg, Some(""))`, (c) Enhance `Builder.deprecated` to not require a source locator

(b) sort of works but you end up with:
```
[deprecated]  (1 calls): calling .suggestName("wire")
```
I think we could make it understand empty source locators to get rid of the `(1 calls)` bit.

Alright we can't call `.toString` but we need to give the user some more information, maybe:
```scala
val name = this._parent.get.name + "." + suggested
Builder.deprecated(
  "calling .suggestName(\"" + suggested + "\") on " + name + " had no effect as it is the same as the automatically given name, this will become an error in 3.6",
  Some("")
)
```

Reply:
how about Some("(unknown)") which is what the deprecated message would do if it couldn't figure out the location

Reply:
Now this is:

Reply:
which I think is OK